ecosystem/indexer-grpc/indexer-grpc-utils/src/file_store_operator/gcs.rs

```diff
@@ -221,28 +243,52 @@ impl FileStoreOperator for GcsFileStoreOperator {
             batch_size % BLOB_STORAGE_SIZE == 0,
             "The number of transactions to upload has to be multiplier of BLOB_STORAGE_SIZE."
         );
-        let mut tasks = vec![];
+        let start_time = std::time::Instant::now();
+        let protobuf_serialized_size = transactions.iter().map(|t| t.encoded_len()).sum::<usize>();
+        let files_to_upload = transactions
+            .into_iter()
+            .chunks(BLOB_STORAGE_SIZE)
+            .into_iter()
+            .map(|i| {
+                let transactions = i.collect_vec();
+                let file_key =
+                    FileEntryKey::new(transactions.first().unwrap().version, self.storage_format);
+                let file_key_string = file_key.to_string();
+
+                let file_entry_builder = FileEntryBuilder::new(transactions, self.storage_format);
+                let file_entry: FileEntry = file_entry_builder
+                    .try_into()
+                    .expect("Failed to serialize file entry");
+                (file_key_string, file_entry.into_inner())
+            })
+            .collect_vec();
```

Comment:
is this the compression step? let's log this as a new step "Compressed data for GCS upload" in counters.rs